irida_staramr_results/parser.py

```diff
@@ -0,0 +1,35 @@
+import yaml
+import logging
+import sys
+
+
+def parse_config(config_file_path):
+    """
+    Parse configuration YAML file provided by the user.
+    It assumes the file name to be "example-config.yml".
+    :param config_file_path:
+    :return config_dict:
+    """
+
+    try:
+        with open(config_file_path, "r") as file:
+            config_info = yaml.load(file, Loader=yaml.FullLoader)
+    except FileNotFoundError as e:
+        logging.error(f"No configuration file found in {config_file_path}.")
+        sys.exit(1)
```

Comment:
When this function was in `cli.py` it was OK to do a `sys.exit` since `cli.py` is the entry point for the application. But now that this function is in it's own module, having it do a hard exit is not nice anymore since a new developer might not understand where the exits are coming from.

Instead, when you would do a `sys.exit`, raise an `Exception` instead, and have the function caller of `parse_config` in `cli.py` have a `try/except` block that does the `sys.exit`.

Reply:
Done. I put parse on its separate module and created custom exception types.

Reply:
Perfect, this looks very clean.